scripts/preprocess_data.py

```diff
@@ -0,0 +1,192 @@
+                fuel = line.split(',')[2]
+                tech = line.split(',')[1]
+            elif line.startswith(start_year):
+                years = line.rstrip().split(' ')[0:]
+                years = [i.strip(':=') for i in years]
+            elif not line.startswith(start_year):
+                values = line.rstrip().split(' ')[1:]
+                mode = line.split(' ')[0]
+                data_out.append(tuple([fuel,tech,mode]))
+                data_all.append(tuple([tech,mode]))
+                for i in range(0,len(years)):
+                    output_table.append(tuple([tech,fuel,mode,years[i],values[i]]))
+        if line.startswith('param OutputActivityRatio'):
+            parsing = True
+
+with open(data_infile, 'r') as f:
+    for line in f:
+        if line.startswith(";"):
+            parsing = False   
+        if parsing:
+            if line.startswith('['):
+                fuel = line.split(',')[2]
+                tech = line.split(',')[1]
+            elif not line.startswith(start_year):
+                mode = line.split(' ')[0]
+                data_inp.append(tuple([fuel,tech,mode]))
+                data_all.append(tuple([tech,mode]))
+        if line.startswith('param InputActivityRatio'):
+            parsing = True
+
+# For TechnologyToStorage and TechnologyFromStorage 
+storage_to = []
+storage_from = []
+
+with open(data_infile) as f:
+    for line in f:
+        if line.startswith(";"):
+            parsing = False   
+        if parsing:
+            if line.startswith('['):
+                tech = line.split(',')[1]
+            elif not line.startswith(mode_list[0]):
+                storage = line.split(' ')[0]
+                values = line.rstrip().split(' ')[1:]
+                for i in range(0,len(mode_list)):
+                    if values[i] != '0':
+                        storage_to.append(tuple([storage,tech,mode_list[i]]))
+                        #data_all.append(tuple([tech,mode_list[i]]))
+        if line.startswith('param TechnologyToStorage'):
+            parsing = True
+
+with open(data_infile) as f:
+    for line in f:
+        if line.startswith(";"):
+            parsing = False   
+        if parsing:
+            if line.startswith('['):
+                tech = line.split(',')[1]
+            elif not line.startswith(mode_list[0]):
+                storage = line.split(' ')[0]
+                values = line.rstrip().split(' ')[1:]
+                for i in range(0,len(mode_list)):
+                    if values[i] != '0':
+                        storage_from.append(tuple([storage,tech,mode_list[i]]))
+                        #data_all.append(tuple([tech,mode_list[i]]))
+        if line.startswith('param TechnologyFromStorage'):
+            parsing = True
+            
+dict_out = defaultdict(list)
+dict_inp = defaultdict(list)
+dict_all = defaultdict(list)
+dict_stt = defaultdict(list)
+dict_stf = defaultdict(list)
+
+for f,t,m in data_out:
```

Comment:
`f,t,m` are not meaningful names. Please use longer names as it makes the code more readable e.g. `fuel, tech, mode`

Reply:
Names are now updated to `fuel, tech, mode`